File: WebApp/BillingApp/src/app/app-layout/product/models/Product.ts
import { ProductSet } from './ProductSet'

export class Product{
    public constructor(init?: Partial<Product>) {
        Object.assign(this, init);
    }
    id:number;
    name:string;
    itemCode:string;
    hsn:string;
    tax:number;
    productSets:ProductSet[]
}

File: BillingAPI/BillingAPI.Models/Product.cs
using System;
using System.Collections.Generic;

namespace BillingAPI.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string ItemCode { get; set; }
        public string HSN { get; set; }
        public int Tax { get; set; }

        public List<ProductSet> ProductSets { get; set; }
    }
}
